alembic/versions/005_add_credit_transaction_provider_reference_constraint.py

```
@@ -0,0 +1,247 @@
+                        provider = 'legacy' if i == 0 else f'legacy_dup_{i+1}'
+                        bind.execute(text("""
+                            UPDATE credit_transactions
+                            SET provider = :provider
+                            WHERE id = :tx_id AND provider IS NULL
+                        """), {"provider": provider, "tx_id": tx_id})
+
+                print("✅ SQLite duplicates handled successfully")
+            else:
+                print("✅ No duplicate reference_ids found")
+
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)